File: leetcode2/0447. Number of Boomerangs.py
class Solution:
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        rep=0
        for k in range(len(points)):
            rep+=self.subnumberOfBoomerangs(points[k],points[:k]+points[k+1:])
        return rep
        
    def subnumberOfBoomerangs(self,point,points):
        rep=0
        
        dictpoints={}
        for p in points:
            k=(p[0]-point[0])**2+(p[1]-point[1])**2
            if k not in dictpoints:
                dictpoints[k]=1
            else:
                dictpoints[k]+=1
        
        for key in dictpoints:
            if dictpoints[key]>1:
                rep+=dictpoints[key]*(dictpoints[key]-1)
        
        return rep

File: archives/C#/0447. Number of Boomerangs.cs
public class Solution {
    public int NumberOfBoomerangs(int[,] points) {
        int rep=0;
        for(int i=0;i<points.GetLength(0);i++){
            for(int j=0;j<points.GetLength(1);j++){
                rep+=GetNumberOfBoomerangs(i,j,points);
            }
        }
        return rep;
    }
    public int GetNumberOfBoomerangs(int i, int j, int[,] points){
        Dictionary<int,int> distancePoint=new Dictionary<int,int>();
        for(int row=i;row<points.GetLength(0);row++){
            for(int column=j;column<points.GetLength(1);column++){
                if(row!=i || column!=j){
                    int tmp=(row-i)*(row-i)+(column-j)*(column-j);
                    if(!distancePoint.ContainsKey(tmp)){
                        distancePoint[tmp]=1;
                    }
                    else{
                        distancePoint[tmp]+=1;
                    }
                }
            }
        }
        int rep=0;
        foreach(KeyValuePair<int,int> pvt in distancePoint){
            rep+=pvt.Value*(pvt.Value-1);
        }
        return rep;
    }
}
